planning/behavior_path_planner/src/utils/avoidance/utils.cpp

```diff
@@ -563,6 +563,50 @@ void fillObjectMovingTime(
   }
 }
 
+void fillAvoidanceNecessity(
+  ObjectData & object_data, const ObjectDataArray & registered_objects, const double vehicle_width,
+  const std::shared_ptr<AvoidanceParameters> & parameters)
+{
+  const auto t = utils::getHighestProbLabel(object_data.object.classification);
+  const auto object_parameter = parameters->object_parameters.at(t);
+  const auto safety_margin =
+    0.5 * vehicle_width + object_parameter.safety_buffer_lateral * object_data.distance_factor;
+
+  const auto one_shot_necessity =
+    (isOnRight(object_data) && std::abs(object_data.overhang_dist) < safety_margin) ||
+    (!isOnRight(object_data) && object_data.overhang_dist < safety_margin);
+
+  const auto id = object_data.object.object_id;
+  const auto same_id_obj = std::find_if(
+    registered_objects.begin(), registered_objects.end(),
+    [&id](const auto & o) { return o.object.object_id == id; });
+
+  if (same_id_obj == registered_objects.end()) {
+    object_data.avoid_required = one_shot_necessity;
+    return;
+  }
+
+  // FALSE -> FALSE or FALSE -> TRUE
+  if (!same_id_obj->avoid_required) {
+    object_data.avoid_required = one_shot_necessity;
+    return;
+  }
+
+  // TRUE -> TRUE
+  if (one_shot_necessity) {
+    object_data.avoid_required = true;
+    return;
+  }
+
+  const auto hysteresis_factor = parameters->safety_check_hysteresis_factor;
+
+  // TRUE -> FALSE (check again with hysteresis factor)
+  object_data.avoid_required =
+    (isOnRight(object_data) &&
```

Comment:
Just a suggestion not necessally to be adapeted
```
bool isAvoidanceRequired(const ObjectData & object_data, double safety_margin, double hysteresis_factor) {
  return (isOnRight(object_data) && std::abs(object_data.overhang_dist) < safety_margin * hysteresis_factor) ||
         (!isOnRight(object_data) && object_data.overhang_dist < safety_margin * hysteresis_factor);
}
```
If this function can be used many times. (only two times in this function but make function small?)

Reply:
Fixed in https://github.com/autowarefoundation/autoware.universe/pull/3880/commits/8c96da4e16e056e9ae957cc0734440a0e60169a5.